server/webapp/WEB-INF/rails/webpack/views/pages/config_repos.tsx

```diff
@@ -25,21 +26,26 @@ import {PluginInfo} from "models/shared/plugin_infos_new/plugin_info";
 import {PluginInfoCRUD} from "models/shared/plugin_infos_new/plugin_info_crud";
 import * as Buttons from "views/components/buttons";
 import {FlashMessage, MessageType} from "views/components/flash_message";
+import {SearchField} from "views/components/forms/input_fields";
 import {HeaderPanel} from "views/components/header_panel";
 import {DeleteConfirmModal} from "views/components/modal/delete_confirm_modal";
-import {Attrs, ConfigReposWidget} from "views/pages/config_repos/config_repos_widget";
+import {ConfigReposWidget, Operations, SearchOperation} from "views/pages/config_repos/config_repos_widget";
 import {EditConfigRepoModal, NewConfigRepoModal} from "views/pages/config_repos/modals";
 import {Page, PageState} from "views/pages/page";
-import {AddOperation, SaveOperation} from "views/pages/page_operations";
+import {AddOperation, RequiresPluginInfos, SaveOperation} from "views/pages/page_operations";
+import * as styles from "./config_repos/index.scss";
 
-interface State extends AddOperation<ConfigRepo>, Attrs<ConfigRepo>, SaveOperation {
+interface State extends AddOperation<ConfigRepo>, SaveOperation, Operations<ConfigRepo>, SearchOperation<ConfigRepo>, RequiresPluginInfos  {
 }
 
 export class ConfigReposPage extends Page<null, State> {
-
   oninit(vnode: m.Vnode<null, State>) {
-    vnode.state.objects     = stream();
-    vnode.state.pluginInfos = stream();
+    vnode.state.pluginInfos    = stream();
+    vnode.state.initialObjects = stream();
+    vnode.state.searchText     = stream();
+    vnode.state.filteredObjects = () => {
+      return stream(vnode.state.initialObjects() ? vnode!.state!.initialObjects()!.filter((o) => o.matches(vnode.state.searchText())) : null);
```

Comment:
We don't need to add `!` after vnode and state. Just `vnode.state.initialObjects()!.filter` should be enough